File: practices/updated_calc.py
# JL 7th updated financial calculator

print("Hello, this is a budget calculator")

def user_info():
    income = int(input("tell me your monthly income: "))
    rent = int(input("tell me your rent: "))
    utilities = int(input("tell me your utilities cost: "))
    groceries = int(input("tell me how much your groceries cost: "))
    transportation = int(input("tell me how much your transportation cost: "))
   
    return income, rent, utilities, groceries, transportation



def budget_calc(income, rent, utilities, groceries, transportation):
    expenses = rent + utilities + groceries + transportation
    savings = income * .1
    spending = income - (expenses + savings)

    rent_percent = (rent/income)*100
    utilities_percent = (utilities/income)*100
    groceries_percent = (groceries/income)*100
    transportation_percent = (transportation/income)*100
    

    print("")
    print("your rent is $" + str(rent) + " which is " + str(round(rent_percent, 2)) + "% of your income")
    print("your utilities are $" + str(utilities) + " which is " + str(round(utilities_percent, 2)) + "% of your income")
    print("your groceries are $" + str(groceries) + " which is " + str(round(groceries_percent, 2)) + "% of your income")
    print("your transportation is $" + str(transportation) + " which is " + str(round(transportation_percent, 2)) + "% of your income")
    print("")
    print("your total expenses are $" + str(expenses))
    print("your savings goal is $" + str(savings))
    print("your spending money is $" + str(spending))

File: c_programming/practices/updatedcalc.c
// JL 7th financial calculator
#include <stdio.h>
#include <math.h>

float get_number(char* type){
    float value;
    printf("how many %s do you have: ", type);
    scanf("%f", &value);
    return value;
}


int main(void){
    float monthlyincome;
    float rent;
    float utilities;
    float groceries;
    float transportation;

   
    printf("what is your monthly income: ");
    scanf("%f", &monthlyincome);
      
    printf("what is your monthly rent: ");
    scanf("%f", &rent);
    
    printf("what is your monthly utilities: ");
    scanf("%f", &utilities);
    
    printf("what is your monthly groceries: ");
    scanf("%f", &groceries);

    printf("what is your monthly transportation: ");
    scanf("%f", &transportation);
   

    printf("Your rent is %.2f and that is %.0f%% of your income.\n", rent, (rent / monthlyincome) * 100.0);
    printf("your utilities is %.2f which is %.0f%% of your income\n", utilities, (utilities / monthlyincome) * 100.0);
    printf("your groceries is %.2f which is %.0f%% of your income\n", groceries, (groceries / monthlyincome) * 100.0);
    printf("your transportation is %.2f which is %.0f%% of your income\n", transportation, (transportation/monthlyincome) * 100.0);
    
    float savings = monthlyincome*0.1;
    
    float spendings = monthlyincome - (rent + utilities + groceries + transportation + savings);
    
    printf("you should save %.2f which is %.0f%% of your income\n", savings, (savings/monthlyincome) * 100.0);
    printf("your spending money is %.2f,\n", spendings);


    return 0;
}